component-library/component-lib/src/lib/form-components/input/input.component.ts

```diff
@@ -218,10 +228,26 @@ export class InputComponent
       );
     }
 
-    //Get the error text when the formControl value changes
-    this.config.formGroup.get(this.config.id)?.statusChanges.subscribe(() => {
+    this.currentStatus = this.config.formGroup.get(this.config.id)?.status || 'DISABLED';
+    switch (this.currentStatus) {
```

Comment:
Definitely weird that a single case switch was used instead of an if/else. Is there a reason for this?

Reply:
Hi Kris, not sure if Alexandre addressed this with you or not. We used it to support future improvements - we'll be able to execute code on 'VALID', 'INVALID', 'DISABLED', etc. If you think this will not ever be used though we could convert it. Let me know.